pysph/cpy/tests/test_jit.py

```diff
@@ -0,0 +1,636 @@
+from math import sin
+import unittest
+import numpy as np
+
+from pytest import importorskip
+
+from ..config import get_config, use_config
+from ..array import wrap
+from ..jit import annotate, get_binop_return_type, AnnotationHelper
+from ..parallel import Elementwise, Reduction, Scan
+
+
+@annotate
+def g(x):
+    return x
+
+
+@annotate
+def h(a, b):
+    return g(a) * g(b)
+
+
+class TestAnnotationHelper(unittest.TestCase):
```

Comment:
Am really glad you've added a lot of tests and are using the Given/When/Then style!  If possible, make sure you have 100% coverage of the annotation helper and other code.  This will be easier when we pull out cpy from pysph but if you can, please measure coverage and make sure it covers all the code.  If this is a pain, we should do this once we pull cpy out.